lib/default.nix

```diff
@@ -298,6 +348,24 @@ let
             )
           );
 
+      # homeConfigurations is defined under legacyPackages instead of at the top level
+      # because Home Manager requires `pkgs` to be instantiated already, which means
+      # knowing the system ahead of time. This means either specifying the system at the
+      # commandline, encoding the system in the path, or using one of the other possible
+      # installable locations.
+      # Nix checks `packages`, `legacyPackages`, then the attribute name at the top level.
+      # If we were to use `packages`, then `nix flake show` would break because it expects
+      # an attrset containing only derivations. `legacyPackages` supports a nested
+      # structure which is necessary for Home Manager.
+      legacyPackages = lib.optionalAttrs (homes != { }) (
+        eachSystem (
+          { pkgs, ... }:
+          {
+            homeConfigurations = lib.mapAttrs (_name: value: value pkgs) homes;
+          }
+        )
+      );
```

Comment:
It would be nice if the output could be attached to the .#packages instead. Having a mix of packages and legacyPackages is confusing.

This should be possible by changing upstream like this:
https://github.com/zimbatm/home-manager/commit/44d0a7e77850a28a8440cf55dff70868bae240b3